app/server.go

```diff
@@ -58,8 +58,8 @@ type CorsWrapper struct {
 }
 
 func (cw *CorsWrapper) ServeHTTP(w http.ResponseWriter, r *http.Request) {
-	if len(*cw.config().ServiceSettings.AllowCorsFrom) > 0 {
-		if utils.OriginChecker(r) {
```

Comment:
Bit of a TOCTTOU error. Could have crashed if the configuration changed between these two lines. Hence the new variable here.